app/models/spree/sale_price.rb

```diff
@@ -2,7 +2,7 @@ module Spree
   class SalePrice < ActiveRecord::Base
 
     belongs_to :price, class_name: "Spree::Price"
-    delegate_belongs_to :price, :currency
+    delegate :currency, to: :price
```

Comment:
You replaced the `delegate_belongs_to` with `delegate`, but you missed the set method `:price=`. I saw this also in other lines.